featureflags/graph/actions.py

```diff
@@ -1,116 +1,43 @@
 
 from aiopg.sa import SAConnection
-from sqlalchemy import select, and_, update, or_
-from prometheus_client import Histogram, Counter
+from sqlalchemy import and_, or_, select, update
 from sqlalchemy.dialects.postgresql import insert
 
-from featureflags import metrics
-from featureflags.utils import select_scalar
-from featureflags.models import Operator, Project, Variable, AuthSession, AuthUser
-from featureflags.models import Flag, Check, LocalIdMap, Condition, Action, Changelog
-
-
-action_time = Histogram(
-    "action_time",
-    "Action latency (seconds)",
-    ["action"],
-    buckets=(0.010, 0.050, 0.100, 1.000, float("inf")),
+from featureflags.graph.constants import AUTH_SESSION_TTL
+from featureflags.graph.metrics import track
+from featureflags.graph.types import (
+    Action,
+    AddCheckOp,
+    AddConditionOp,
+    Changes,
+    DirtyProjects,
+    LocalId,
 )
-
-action_errors = Counter(
-    "action_errors",
-    "Action errors count",
-    ["action"],
+from featureflags.graph.utils import update_map
+from featureflags.models import (
+    AuthSession,
+    AuthUser,
+    Changelog,
+    Check,
+    Condition,
+    Flag,
+    LocalIdMap,
+    Operator,
+    Project,
+    Variable,
 )
+from featureflags.services.auth import UserSession, auth_required
+from featureflags.services.ldap import BaseLDAP
+from featureflags.utils import select_scalar
 
 
-def measure_action(func):
-    name = func.__name__
-    func = metrics.wrap(action_time.labels(name).time())(func)
-    func = metrics.wrap(action_errors.labels(name).count_exceptions())(func)
-    return func
-
-
-SESSION_TTL = timedelta(days=14)
-
-
-def _update_ids_map(ids_map, ids):
-    ids_map = ids_map.copy()
-    ids_map.update(ids)
-    return ids_map
-
-
-class AccessError(Exception):
-    pass
-
-
-session_var = ContextVar("session")
-
-
-@contextlib.asynccontextmanager
-async def with_session(session):
-    try:
-        session_var.set(session)
-        yield
-    finally:
-        session_var.set(None)
-
-
-def auth_required(func):
-    async def wrapper(*args, **kwargs):
-        session = session_var.get()
-        if not session.is_authenticated:
-            raise AccessError("User is not authenticated")
-        return await func(*args, **kwargs)
-
-    return wrapper
-
-
-class DirtyProjects:
-    def __init__(self):
-        self.by_flag = set()
-        self.by_variable = set()
-
-
-class Changes:
-    _data: Dict[UUID, List[Action]]
-
-    def __init__(self):
-        self._data = defaultdict(list)
-
-    def add(self, flag_id: UUID, action: Action):
-        self._data[flag_id].append(action)
-
-    def get_actions(self):
-        return list(self._data.items())
-
-
-@dataclass
-class LocalId:
-    scope: str
-    value: str
-
-    def __hash__(self):
-        return hash((self.scope, self.value))
-
-
-async def gen_id(local_id: LocalId, *, db: SAConnection):
+async def gen_id(local_id: LocalId, *, db_connection: SAConnection) -> UUID:
```

Comment:
nit: it is not a big deal but `db_connection` is way to long variable. If there is not other connections such as redis or else, I would suggest to use `db` or `conn` as a var name.